solr/solrj/src/java/org/apache/solr/common/MapWriter.java

```diff
@@ -70,8 +74,7 @@ public MapWriter.EntryWriter put(CharSequence k, Object v) {
     writer.endObject();
   }
 
-  void writeMap(EntryWriter ew) throws IOException;
-
+  @Deprecated
```

Comment:
I kind of wish we always provide a bit more detail on these `@Deprecate` tags.  I like when I see a "since" as it let's me know when the code can be deleted.   I like when there is a "use X instead" comment.

I did a pass the other week around deprecated code and a lot it wasn't clear what would trigger the deleting of it..  Or how long it had been deprecated.

Reply:
for an obscure method; I think we shouldn't burden ourselves with overly helping likely non-existent users of such.  Personally I do plenty of code sleuthing to identify when lines of code are added/removed like a Deprecate or whatever.

Reply:
> I did a pass the other week around deprecated code and a lot it wasn't clear what would trigger the deleting of it.. Or how long it had been deprecated.

Curious; in that instance, why did you care?  In this case, it's just some small utility method that any caller can do themselves.  Thus the deprecation rationale is not something I think any user would find interesting _for this method_.

Reply:
I was looking at code in 10 that was marked deprecated so I could remove it.  I'd love to see a release of 10 that has NO deprecation tags in it, which may be a pipe dream...  I was hoping to see some indication that would tell me what methods to be used in place for example.